src/utilities.cpp

```diff
@@ -88,8 +88,8 @@ void pasteFilesFromClipboard(const Fm::FilePath& destPath, QWidget* parent) {
     if(paths.empty() && data->hasUrls()) {
         // The KDE way
         paths = Fm::pathListFromQUrls(data->urls());
-        QByteArray cut = data->data("x-kde-cut-selection");
-        if(!cut.isEmpty() && cut.at(0) == '1') {
+        QByteArray cut = data->data(QStringLiteral("application/x-kde-cutselection"));
+        if(!cut.isEmpty() && cut.at(0) == QLatin1Char('1')) {
```

Comment:
@vc-01 
 All new code should compile with:
``` cmake
add_definitions(
    -DQT_USE_QSTRINGBUILDER
    -DQT_NO_CAST_FROM_ASCII
    -DQT_NO_CAST_TO_ASCII
    -DQT_NO_URL_CAST_FROM_STRING
    -DQT_NO_CAST_FROM_BYTEARRAY
)
```
From the doc:

> If you defined QT_NO_CAST_FROM_ASCII and QT_NO_CAST_TO_ASCII, as explained in the QString documentation, you will need to explicitly call fromLatin1(), or use QLatin1Char, to construct a QChar from an 8-bit char, and you will need to call toLatin1() to get the 8-bit value back.

But you are right, the code I proposed doesn't work. `QChar::QChar(char ch)` is private.
``` c++
QChar::fromLatin1(c.at(0))
```
does the job